homeassistant/components/pyload/coordinator.py

```diff
@@ -0,0 +1,73 @@
+"""Update coordinator for pyLoad Integration."""
+
+from datetime import timedelta
+import logging
+
+from pyloadapi import (
+    CannotConnect,
+    InvalidAuth,
+    ParserError,
+    PyLoadAPI,
+    StatusServerResponse,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=20)
+
+
+class pyLoadData(StatusServerResponse):
+    """Data from pyLoad."""
+
+    free_space: int
+    version: str
```

Comment:
```suggestion
@dataclass(kw_only=True)
class PyLoadData:
    """Data from pyLoad."""

    status: StatusServerResponse
    free_space: int
    version: str
```

Reply:
i like that solution, didn't think of that ^^

Reply:
Ok, one thing i just noticed, this would complicate the value retrieval for the sensors. 

This works for status
`self.coordinator.data.status[self.entity_description.key]`
but for retrieving for example free_space we need to access it differently
😕